base/complex.jl

```diff
@@ -533,8 +534,12 @@ function expm1(z::Complex)
             Complex(erm1, zi)
         else
             er = erm1+one(erm1)
-            wr = isfinite(er) ? erm1 - 2.0*er*(sin(0.5*zi))^2 : er*cos(zi)
-            Complex(wr, er*sin(zi))
+            if isfinite(er)
+                wr = erm1 - 2 * er * (sin(convert(Tf, 0.5) * zi))^2
```

Comment:
I'm just trying to not change the result for machine types and being conservative about this transformation. I assume this transformation is exact for Float16, Float32 and Float64 (though floating point corner cases frequently surprises me...)? If that's the case then I think we can change this to `/ 2`.